docs/Practices/hub/examples.md

```diff
@@ -0,0 +1,113 @@
+# Examples
+
+## Hub Directory Structure
+```
+// WIP
+bounded-contexts
+  some-bc-1
```

Comment:
What about a /lib directory as well?

Reply:
Within a slice as long as, you are following the existing conventions, I don't think it really matters how you structure your code past those conventions.

So `lib`, `fns`, loose files with good (*subjective I know*) names are all valid if they help solve the problem.

The intent the existing conventions within a slice existed was so you could easily find where a command or query entrypoint was, where it integrated with the database, where it defined slice specific domain code/derivatives of shared domain code (if you even need to do this), and where the code exposing commands outside of the service live (http request handlers and maybe event handlers one day.

The hope is to find a sweet spot between Vertical Slices and Clean Architecture where we can gain the primary benefits while minimizing the costs associated with both.

This all being said, I'm not opposed to a convention that adds a `lib` dir if it will help us in this task. If you are proposing a convention that utilizes a `lib` dir then could you respond with what you see this lib dir containing and the rules around it?

Reply:
I see a `lib` as the general code that is needed for the whole Context to work. Helper methods that are not tied to the outcome of a single action but what makes the code dry.

Reply:
Ah I got confused, you did indeed comment within a given BC and not a slice.

Yea this makes sense. Depending on how generic these helpers are I could see `lib` existing outside of the bounded context directory. Essentially a lot of helpers aren't domain specific - they just add capabilities that TS/JS/ES is missing.

We have this currently, it's our `util` directory. The downside is that this `util` dir has stuff that is domain specific since it existing before we started adding boundaries. A good refactor might be to create a `lib` directory and extract out all generic stuff into it. This is best as a different set of PRs that includes changes to this doc + to Hub.

A really key point of slices though is reducing coupling between slices. We already have some coupling through types derived from a shared domain layer, repository, and data (soon to be resources) layers. I wonder what else could be shared at the BC level that doesn't just belong in the afformentioned `util` - ideally refactored to `lib` - dir.

Maybe we live for a time without a BC level lib dir and then approach this topic as code to be extracted presents itself? I think this commend by Jimmy in response to a question on his article about Vertical Slices is a good signal for how to think about sharing code between slices. http://disq.us/p/1y542go